src/ParametersHandler/YarpImplementation/include/BipedalLocomotion/ParametersHandler/YarpImplementation.tpp

```diff
@@ -72,6 +72,8 @@ void YarpImplementation::setParameterPrivate(const std::string& parameterName, c
 
         m_lists[parameterName] = std::make_shared<YarpImplementation>(yarpValue);
     }
+
+    m_paramNames.insert(parameterName);
```

Comment:
``parameterName`` can be both the name of the parameter and the name of the group. See two lines above.

Reply:
I knew I didn't get correctly the logic of the Bottle. I'll try again with your suggestion of https://github.com/dic-iit/bipedal-locomotion-framework/pull/134#discussion_r518584137. I'm not a big fan either to introduce new data structures but it looked the fastest way.

As commented elsewhere, I believe that the storage based on Bottle could be removed and substituted by the implementation of the `StdImplemention`. In this case, all this Bottle logic could be removed.